docs/fastdds/ros2/Discovery_server/ros2_discovery_server.rst

```diff
@@ -0,0 +1,357 @@
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811;127.0.0.1:11888"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker_1
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811;127.0.0.1:11888"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener_1
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker_2
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER=";127.0.0.1:11888"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener_2
+
+We should see how *Listener 1* is receiving double messages while *Listener 2* is in a different
+partition of *Talker 2* and so it does not listen it.
+
+.. note::
+
+    Once two endpoints know each other, they do not need the server network between them to
+    listen to each other messages.
+
+
+
+Compare Discovery Server with Simple Discovery
+-----------------------------------------------
+
+In order to compare the ROS2 execution using *Simple Discovery* or *Discovery Service*, we provide two scripts that
+execute a talker and many listeners and analyze the network traffic during this time. For this you will need to install
+``tshark`` on your system.
+
+However, this functionality are references for advance purpose and we are going to leave its studying to the user.
+
+:download:`bash network traffic generator <generate_discovery_packages.bash>`
+
+:download:`python3 graph generator <discovery_packets.py>`
+
+Run the bash script with the *setup* path to source ROS2 as argument.
+This will generate the traffic trace for simple discovery.
+Executing the same script with second argument ``SERVER``, it will generates the trace for service discovery.
+
+.. note::
+
+    Depending on your configuration of ``tcpdump`` this script may require ``sudo`` privileges to read traffic across
+    your network device.
+
+After both executions are done, run the python script to generates a graph similar to the one below:
+
+.. code-block:: console
+
+    $ sudo bash generate_discovery_packages.bash ~/ros2_foxy/install/local_setup.bash
+    $ sudo bash generate_discovery_packages.bash ~/ros2_foxy/install/local_setup.bash SERVER
+    $ python3 discovery_packets.py
+
+.. image:: /01-figures/fast_dds/discovery/discovery_packets.svg
+    :align: center
+
+This graph is the result of a  is a specific example, the user can execute the scripts and watch their own results.
+It can easily be seen how the network traffic is reduced when using *Discovery Service*.
+
+The reduction in traffic is a result of avoiding every node announcing itself and waiting a response from every other
+node in the net.
+This creates a huge amount of traffic in large architectures.
+This reduction from this method increases with the number of Nodes, making this architecture more scalable than the
+simple one.
```

Comment:
I think it's worth mentioning in this section that, since v2.0.2, the traffic will be massively reduced when nodes do not share topics, which is not the case here.